### ISSUE TITLE
Multi-key primary keys are not supported

### ISSUE THREAD

**Issue by LucaCappelletti94** [gluesql/gluesql#1514]
Hi all,

I just received the following error when trying to create a table that has a composite primary key.

Am I correct in assuming that multiple primary keys are not supported at this time?

What would be necessary to add support for this feature? I am willing to try and add it.

```
Translate(UnsupportedConstraint("PRIMARY KEY (table_id, user_id)"))
```

**Comment by panarch** [comment]
Yes, you are correct that composite primary keys are not supported at this time. Implementing this feature requires significant changes in the core areas, including the AST, planner, and execution layer. This can be a challenging issue, especially for someone just starting out. However, feel free to give it a try, and if you have any questions, leave a comment, and we will support you.

There are multiple storage options, but I recommend focusing on the memory storage first and ignoring the other storage types for now. If you can successfully implement the feature for one storage, it will not be too difficult to extend it to the others.

**Comment by LucaCappelletti94** [comment]
Ok! Which files should I check out first?

**Comment by LucaCappelletti94** [comment]
Fortunately, in the SQL parser, I see that the multi-column primary keys are already supported. Hopefully, no edits will be needed there.

**Comment by LucaCappelletti94** [comment]
In this enum, shouldn't there be more options? 

https://github.com/gluesql/gluesql/blob/25309dd15a50ee061e714576079b2cd8c1fcca9c/core/src/ast/mod.rs#L43-L46

**Comment by LucaCappelletti94** [comment]
Where in the code is the unique constraint enforced currently?

**Comment by panarch** [comment]
> In this enum, shouldn't there be more options?
> 
> https://github.com/gluesql/gluesql/blob/25309dd15a50ee061e714576079b2cd8c1fcca9c/core/src/ast/mod.rs#L43-L46

it can have more, but not yet supported.

**Comment by panarch** [comment]
> Where in the code is the unique constraint enforced currently?

here. it includes both clustered (primary key) and non-clustered index validation codes. it would be enough to focus on primary key related part.
https://github.com/gluesql/gluesql/blob/main/core/src/executor/validate.rs

**Comment by leejw51** [comment]
0.15.0 is fine for `Multi-key primary key`, is it ok to use multikey primary key on 0.15.0?

**Comment by devgony** [comment]
> 0.15.0 is fine for `Multi-key primary key`, is it ok to use multikey primary key on 0.15.0?

We don't support the feature yet, can you share your schema? (or CREATE TABLE SQL)
